src/ExtraSessionTrait.php

```diff
@@ -76,13 +90,154 @@ public function iWaitForCssElementBeingInvisible($element, $seconds)
+     * @param string $element
+     *
+     * @throws ExpectationException
      */
-    public function scrollTo($x, $y)
+    public function iWaitUntilCssElementBeingVisible($element)
     {
-        $this->getSession()->executeScript("(function(){window.scrollTo($x, $y);})();");
+        $this->spin(function (MinkContext $context) use ($element) {
+            /** @var \Behat\Mink\Element\NodeElement[] $elements */
+            $elements = $context->getSession()->getPage()->findAll('css', $element);
+            if (\count($elements) === 0 || $element === null) {
+                return false;
+            }
+            foreach ($elements as $element) {
+                if ($element->isVisible()) {
+                    return true;
+                }
+            }
+            return false;
+        }, 'iWaitUntilCssElementBeingVisible::'.$element);
+    }
+
+    /**
+     * Wait until the given css element being masked
+     *
+     * @param string $element
+     *
+     * @throws ExpectationException
+     */
+    public function iWaitUntilCssElementBeingInvisible($element)
+    {
+        $this->spin(function (MinkContext $context) use ($element) {
+            $element = $context->getSession()->getPage()->find('css', $element);
+            if ($element === null) {
+                return false;
+            }
+            return !$element->isVisible();
+        }, 'iWaitUntilCssElementBeingInvisible::'.$element);
+    }
+
+    /**
+     * Wait until the page contains given text
+     *
+     * @param string $text
+     *
+     * @throws ExpectationException
+     */
+    public function iWaitUntilPageContains($text)
+    {
+        $this->spin(function (MinkContext $context) use ($text) {
+            // Assertion throw exception if not correct, nothing if correct
+            try {
+                $context->assertSession()->pageTextContains($text);
+                return true;
+            } catch (ResponseTextException $e) {
+                return false;
+            }
+        }, 'iWaitUntilPageContains::'.$text);
+    }
+
+    /**
+     * Wait until the page not contains given text
+     *
+     * @param string $text
+     *
+     * @throws ExpectationException
+     */
+    public function iWaitUntilPageNotContains($text)
+    {
+        $this->spin(function (MinkContext $context) use ($text) {
+            // Assertion throw exception if not correct, nothing if correct
+            try {
+                $context->assertSession()->pageTextNotContains($text);
+                return true;
+            } catch (ResponseTextException $e) {
+                return false;
+            }
+        }, 'iWaitUntilPageNotContains::'.$text);
+    }
+
+    /**
+     * Click on the element matching given selector
+     *
+     * @Given /^I click on element "(?P<selector>[^"]*)"$/
+     *
+     * @param string $selector
+     *
+     * @throws ElementNotFoundException
+     */
+    public function iClickOnCssElement($selector)
```

Comment:
Already exists `clickElement` you remove in this PR, that is a BC break for the projects using this lib